File: guides/voice/gather-dtmf-tones-guide/gather-example-step-2.1/example.5.x.cs
// In Package Manager, run:
// Install-Package Twilio.AspNet.Mvc -DependencyVersion HighestMinor

using System.Web.Mvc;
using Twilio.AspNet.Mvc;
using Twilio.TwiML;

public class VoiceController : TwilioController
{
	[HttpPost]
	public ActionResult Gather(string digits)
	{
		var response = new VoiceResponse();

		// If the user entered digits, process their request
		if (!string.IsNullOrEmpty(digits))
		{
			switch (digits)
			{
				case "1":
					response.Say("You selected sales. Good for you!");
					break;
				case "2":
					response.Say("You need support. We will help!");
					break;
				default:
					response.Say("Sorry, I don't understand that choice.").Pause();
					response.Redirect("/voice");
					break;
			}
		}
		else
		{
			// If no input was sent, redirect to the /voice route
			response.Redirect("/voice");
		}

		return TwiML(response);
	}
}

File: guides/voice/gather-dtmf-tones-guide/gather-example-step-1/example.5.x.php
<?php
// Create a route that will handle Twilio webhook requests, sent as an
// HTTP POST to /voice in our application
require_once '/path/to/vendor/autoload.php';

use Twilio\TwiML\VoiceResponse;

// Use the Twilio PHP SDK to build an XML response
$response = new VoiceResponse();

// If the user entered digits, process their request
if (array_key_exists('Digits', $_POST)) {
    switch ($_POST['Digits']) {
    case 1:
        $response->say('You selected sales. Good for you!');
        break;
    case 2:
        $response->say('You need support. We will help!');
        break;
    default:
        $response->say('Sorry, I don\'t understand that choice.');
    }
} else {
    // If no input was sent, use the <Gather> verb to collect user input
    $gather = $response->gather(array('numDigits' => 1));
    // use the <Say> verb to request input from the user
    $gather->say('For sales, press 1. For support, press 2.');

    // If the user doesn't enter input, loop
    $response->redirect('/voice');
}

// Render the response as XML in reply to the webhook request
header('Content-Type: text/xml');
echo $response;
